java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -470,6 +471,42 @@ public static HostColumnVector fromStrings(String... values) {
     });
   }
 
+
+  /**
+   * Create a new vector from the given values.  This API supports inline nulls, but it is inefficient.
+   * Notice all input BigDecimals should share same scale.
+   */
+  public static HostColumnVector fromBigDecimals(BigDecimal... values) {
+    // 10000 is used as a default value of scale since no valid scale exceeds 19.
+    int scale = 10000;
+    DType.DTypeEnum typeId = DType.DTypeEnum.DECIMAL32;
+    for (BigDecimal v : values) {
+      if (v == null) {
+        continue;
+      }
+      if (scale == 10000) {
+        scale = v.scale();
+      }
+      // check scale consistence
+      assert v.scale() == scale : "Inconsistent decimal scale: " + scale + " vs " + v.scale();
+      // determine the underlying typeId according to maximum precision of inputs.
+      if (v.precision() > DType.DECIMAL64_MAX_PRECISION) {
+        throw new AssertionError("Decimal precision exceeds max precision: " + DType.DECIMAL64_MAX_PRECISION);
+      }
+      if (v.precision() > DType.DECIMAL32_MAX_PRECISION && typeId == DType.DTypeEnum.DECIMAL32) {
+        typeId = DType.DTypeEnum.DECIMAL64;
+      }
+    }
+    DType decType;
+    if (scale != 10000) {
+      decType = DType.create(typeId, -scale);
+    } else {
+      // If input is empty or all input numbers are null, set the default type.
```

Comment:
I'm not a fan of this as the only way to handle things.  I'm also not a fan of the double-loop to sanity-check the scale value.  How about adding a new `fromDecimals(int scale, BigDecimal... values)` method where the user can specify the desired scale explicitly?  This enables a way for the user to specify the expected type even if all the decimal values end up being null.  Then `fromDecimals(BigDecimal.. values)` can avoid the sentinel value mess and extra loop through the values and simply check if there's zero or more than zero values.  For the former, it can select 0 as it does now.  For the latter, it just needs to grab the scale from the first value.  Then it can call `fromDecimals(int scale, BigDecimal... values)` to do the rest.

Reply:
While implementing the additional method `fromDecimals(int scale, BigDecimal... values)`, I found perhaps there is a more concise approach to avoid sentinel value and (explicit) extra loop.
```java
    // Try to fetch the element with max precision. Fill with ZERO if inputs is empty.
    BigDecimal maxPrecisionDec = Arrays.stream(values).filter(Objects::nonNull)
        .max(Comparator.comparingInt(BigDecimal::precision))
        .orElse(BigDecimal.ZERO);
    // Select appropriate underlying type according to max precision.
    DType.DTypeEnum typeId = DType.DTypeEnum.DECIMAL32;
    if (maxPrecisionDec.precision() > DType.DECIMAL64_MAX_PRECISION) {
      throw new AssertionError("Decimal precision exceeds max precision: " + DType.DECIMAL64_MAX_PRECISION);
    } else if (maxPrecisionDec.precision() > DType.DECIMAL32_MAX_PRECISION) {
      typeId = DType.DTypeEnum.DECIMAL64;
    }
    // Because all scales are same, just take scale of maxPrecisionDec.
    int scale = -maxPrecisionDec.scale();
    return build(DType.create(typeId, scale), values.length, (b) -> b.appendBoxed(values));
```
Does it look good to you, @jlowe ?